api/catalog_test.go

```diff
@@ -220,6 +220,73 @@ func TestAPI_CatalogServices_NodeMetaFilter(t *testing.T) {
 	})
 }
 
+func TestAPI_CatalogServices_NodeMetaViaFilter(t *testing.T) {
+	t.Parallel()
+	meta := map[string]string{"somekey": "somevalue", "synthetic": "true"}
+	c, s := makeClientWithConfig(t, nil, func(conf *testutil.TestServerConfig) {
+		conf.NodeMeta = meta
+		conf.NodeName = "foobar"
+	})
+	defer s.Stop()
+
+	catalog := c.Catalog()
+	retry.Run(t, func(r *retry.R) {
```

Comment:
Add a comment describing the retry like `// Make sure we get the service back when filtering by filter expression`